src/github.js

```diff
@@ -36,26 +36,69 @@ async function createStatusCheck(repoToken, reportData, markupData, conclusion)
   }
 }
 
-async function createPrComment(repoToken, markupData) {
-  try {
-    if (github.context.eventName != 'pull_request') {
-      core.info('This event was not triggered by a pull_request.  No comment will be created.');
-    }
+async function createComment(octokit, markupData) {
+  core.info(`Creating PR Comment...`);
+  const response = await octokit.rest.issues.createComment({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
+    issue_number: github.context.payload.pull_request.number,
+    body: markupData
+  });
 
-    core.info(`Creating PR Comment...`);
-    const octokit = github.getOctokit(repoToken);
+  if (response.status !== 201) {
+    core.setFailed(`Failed to create PR comment. Error code: ${response.status}`);
+  } else {
+    core.info(`Created PR comment: ${response.data.id} with response status ${response.status}`);
+  }
+}
+
+async function createOrUpdateComment(octokit, markupData) {
+  const commentsResponse = await octokit.rest.issues.listComments({
+    issue_number: github.context.issue.number,
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo
+  });
+
+  if (commentsResponse.status !== 200) {
+    core.setFailed(`Failed to list PR comments. Error code: ${commentsResponse.status}`);
+    return;
+  }
+
+  const prefixedMarkupData = '<!-- im-open/process-dotnet-test-results -->' + markupData;
```

Comment:
You probably need to add a newline here between the comment and the markupData, otherwise the first item in the markup data (a heading) won't render properly.

Reply:
I've added a newline